app/src/utils/aws/website.ts

```diff
@@ -0,0 +1,98 @@
+import { useAppStore } from "~/state/store";
+import { v4 as uuidv4 } from "uuid";
+import { inverseDatePrefix } from "~/utils/aws/utils";
+
+export const uploadDatasetEntryFileToS3 = async (projectId: string, file: File) => {
+  const { api } = useAppStore.getState();
+
+  if (!api) throw Error("api not initialized");
+
+  // base name without extension
+  const basename = file.name.split("/").pop()?.split(".").shift();
+  if (!basename) throw Error("basename not found");
+
+  const blobName = `${inverseDatePrefix()}-${basename}-${uuidv4()}-uploaded.jsonl`;
+
+  // Step 1: Initiate multipart upload to get the upload ID
+  const { uploadId } = await api.client.datasets.initiateMultipartUpload.query({
+    projectId,
+    filename: blobName,
+  });
+
+  // Step 2: Split the file into parts and get presigned URLs for each part
+  const PART_SIZE = 5 * 1024 * 1024; // 5MB for each part
+  const fileSize = file.size;
+  const NUM_PARTS = Math.ceil(fileSize / PART_SIZE);
+  const partUploadPromises = [];
+
+  for (let partNumber = 1; partNumber <= NUM_PARTS; partNumber++) {
+    const start = (partNumber - 1) * PART_SIZE;
+    const end = Math.min(start + PART_SIZE, fileSize);
+    const part = file.slice(start, end);
+
+    // Get a presigned URL for this part
+    const { presignedUploadUrl } = await api.client.datasets.getPresignedUploadUrlForPart.query({
+      projectId,
+      filename: blobName,
+      partNumber,
+      uploadId: uploadId as string,
+    });
+
+    // Step 3: Upload the part
+    const uploadPartPromise = await fetch(presignedUploadUrl, {
```

Comment:
I'm a bit confused here. You `await`, this, but then below also have an `await Promise.all(partUploadPromises)`. Is one of those `await`s unnecessary?

Reply:
I was facing issues with the ETag and parts values while fetching. Removed the await on fetch so the array is full of promises now. Also removed the unused `completeMultipartUploadResponse` variable. Thanks for catching! Will clean up rest of the code too.